main.cpp

```diff
@@ -213,191 +334,379 @@ struct Computer
         float readSpeedMBs = 101.1f;
         float writeSpeedMs = 74.2f;
 
-        int readData(int address); // returns the value at that address
+        int readData(unsigned int address); 
         
-        bool writeData(int address, int data); // return true if the write succeeded
+        bool writeData(unsigned int address, unsigned int data); 
 
-        bool parkHeads(); // return true if the heads are parked
+        bool parkHeads(); 
     };
 
-    // 3 things it can do:
-    // boot up
-    bool bootUp(Drive systemDrive); // returns true if the boot is successful
+    bool bootUp(Drive systemDrive); 
 
-    // run an program on given path
-    bool runProgram(Drive programDrive, std::string path); // returns true if the program is started successfully
+    bool runProgram(Drive programDrive, std::string path); 
 
-    // crash
-    void crash();    
+    bool crash();    
 };
 
+int Computer::Drive::readData(unsigned int address)
+{
+    return static_cast<int> (address) * rand();
+}
+
+bool Computer::Drive::writeData(unsigned int address, unsigned int data)
+{
+    bool writeFailed = false;
+    
+    srand(address*data); 
+    if ((rand() % 1000000) == 0) // Very bad harddrive!
+    {
+        writeFailed = true;
+    }
+    
+    return writeFailed;
+}
+
+bool Computer::Drive::parkHeads()
+{
+    bool isParked = true;
+    
+    if ((rand() % 1000000) == 0)
+    {
+        isParked = false;
+    }
+
+    return isParked;
+}
+
+bool Computer::bootUp(Drive systemDrive)
+{
+    bool isBooted = true;
+    int data;
+    
+    data = systemDrive.readData(static_cast<unsigned int>(rand()));
```

Comment:
Casting has not been covered yet.  
Remove this cast.
Change your code so that no warnings are generated **without** using casting.
- [ ] revise